src/app/app_wrangling.R

```diff
@@ -0,0 +1,204 @@
+    # subset based on user selection
+    func_df_out <- func_df_out[call_bool_series_or(data, col_, list_), ]
+    # call form group to add group column
+    func_df_out <- form_group(func_df_out, col_, list_)
+    # remove all entries that aren't part of a group
+    fucn_df_out <- filter(func_df_out, !is.na(group))
+    return(fucn_df_out)
+}
+
+
+# used in form group map call to check what should be assigned
+form_group_helper <- function(data, list_) {
+    if (length(data) == 0) {
+        return(NA)
+    }
+    else if (length(list_) > 1 & all(data)) {
+        return("All Selected")
+    }
+    else {
+        return(list_[data])
+    }
+
+}
+
+
+# form groupings based on user selection from `call_boardgame_radio`
+form_group <- function(data, col_, list_) {
+
+    # check if values contain the user input
+    check <- check_list(data[[col_]], list_)
+
+    # assign correct value
+    output <- map(check, ~ (form_group_helper(.x, {{list_}}))) %>%
+        unlist()
+
+    # add new column to dataframe
+    func_df_out <- data %>%
+        mutate(group = output)
+
+    return(func_df_out)
+}
+
+
+# provides group counts after `call_boardgame_radio()` is used
+count_group <- function(data) {
+
+    # create dataframe with counts for each category
+    func_df_out <- data %>%
+        count(year_published, group) %>%
+        pivot_wider(names_from = group, values_from = n)
+
+    # if `All Selected` exists, add counts to other categories
+    if ("All Selected" %in% names(func_df_out)) {
+        # remove na values from `All Selected`
+        temp1 <- func_df_out$`All Selected` %>% replace_na(0)
+        # select category columns
+        temp2 <- func_df_out %>% select(-year_published, -`All Selected`)
+        # add `All Selected to each column`
+        new_df_out <- map_df(temp2, ~ (. + temp1))
+        # create new output df by adding removed columns
+        func_df_out <- new_df_out %>%
+            mutate(year_published = func_df_out$year_published,
+                   `All Selected` = func_df_out$`All Selected`) %>%
+            select(year_published, everything())
+    }
+
+    return(func_df_out)
+}
+
+
+# return top 5 values based on the average for a given column
+call_boardgame_top <- function(data, col_, year_in, year_out) {
+    func_df_out <- data %>%
+        filter(year_published >= year_in, year_published <= year_out) %>%
+        unnest(.data[[col_]]) %>%
+        # note, groupby did not like being passed curly curly
+        group_by(.data[[col_]]) %>%
+        summarize(average = mean(average_rating)) %>%
+        arrange(desc(average)) %>%
+        slice_head(n = 5)
+
+    return(func_df_out)
+}
+
+
+# return unique values to populate dropdowns
+subset_data <- function(data, col_ = "category") {
+    func_df_out <- data %>% unnest(.data[[col_]])
+
+    unique_out <- map(func_df_out[[col_]], ~ (unlist(.x))) %>%
+        unlist() %>%
+        unique() %>%
+        na.omit()
+
+    return(list(unique_out))
```

Comment:
i changed `return(unique_out)` to `return(list(unique_out))`

If i dont make the change it looks like i only get one element back

i.e the drop down boxes on the first tab only have 1 thing listed